### DISCUSSION TITLE
Are there theme types in INTPUT / DYNAMIC-INPUT ?

### DISCUSSION THREAD

**Discussion by TheFreeman** [tusen-ai/naive-ui#2170] [Q&A]
Hi,

there are theme types for BUTTONS for example.
![image](https://user-images.githubusercontent.com/681425/149344255-f1bb84fe-666a-4fff-800a-3c6eb3e2dc5f.png)

So ist there any "type" theme property for INPUT?
I saw in the theme vars, that there are set colors for that types.
But how can i use it? For example on input-errors?

kr

**Comment by 07akioni** [comment]
https://www.naiveui.com/en-US/os-theme/docs/customize-theme#Use-the-peers-vars

**Reply by 07akioni** [reply]
```
const themeOverrides = {
  DynamicInput: {
    peers: {
       Input: xxx,
       Button: xxx
    }
  }
}
```

**Comment by 07akioni** [comment]
https://github.com/TuSimple/naive-ui/blob/main/src/dynamic-input/styles/light.ts

**Comment by 07akioni** [comment]
![image](https://user-images.githubusercontent.com/18677354/149353599-7aa2a011-46db-4d52-b003-029b6fa0658d.png)

**Comment by TheFreeman** [comment]
Thank you!

**Comment by TheFreeman** [comment]
Your answers are not really an answer to my question.

Your documentation...
"In many cases, another component will be reused inside a component, so the theme variable of peers appears.

The theme variables related to peers have not been exposed yet. Use GlobalThemeOverrides to view the peers variables of the corresponding component.

The specific available peers will be updated later."
...is not really good to understand, if you are not the developer of the framework. :-(

I dont understand how to manage your answers regarding to my first post.

**Reply by 07akioni** [reply]
Sorry, I don't fully understand you question.

Are you asking if there are error type for input?

You can't directly set an error status on input component. It must be combined with form-item.

We'll optimize it later to make status directly work on input component (or select component...)

There are some theme variables related to error status for input inside form item with corresponding status.

**Reply by 07akioni** [reply]
![image](https://user-images.githubusercontent.com/18677354/149470790-395cc5d4-37a8-4a94-9fea-99c69aaac316.png)

**Reply by 07akioni** [reply]
You are right, the doc is not easy to understand. We'll try to optimize the docs. Sorry for that. I'm not a native English speaker. Feel free to point out what's hard to understand and we'll finally make it clear.

**Reply by Uhasith** [reply]
where can i find this list in the DOC? @07akioni

**Comment by TheFreeman** [comment]
What exactly is the idea behind "peers" and how exactly does it work?
I don´t understand it at all.

(and by the way, i´m also no native english  speaker) :-)

**Reply by 07akioni** [reply]
> What exactly is the idea behind "peers" and how exactly does it work? I don´t understand it at all.
> 
> (and by the way, i´m also no native english speaker) :-)

Peers is some components that're reused in another component. We don't want same vars to be repeated many times in the whole component library.

For example, select is composed of 2 component. Cascader is composed of 3 components.

```
<select> // use select theme vars
  <internal-selection /> // Peers of select, use select.peers.InternalSelection theme vars
  <internal-select-menu /> // Peers of select, use select.peers.InternalSelectMenu theme vars
</select>
```

```
<cascader>  // use cascader theme vars
  <cascader-menu />  // use cascader theme vars
  <internal-selection /> // Peers of cascader, use cascader.peers.InternalSelection theme vars
  <internal-select-menu /> // Peers of cascader, use cascader.peers.InternalSelectMenu theme vars
</cascader>
```

**Comment by TheFreeman** [comment]
Thank you for implement it to the newest version! :-)
It would be great , if other form-components will follow. :-)
